closed/custom/Docs.gmk

```diff
@@ -103,18 +100,220 @@ OPENJ9_JAVADOC_SCRIPT_BODY = \
+# javadoc is not generated for the classes  which are replaced by openj9
+# but are functionally identical to the openjdk implementations
+# (e.g. java.lang.String), since the openjdk javadoc is still applicable.
+# 
+# The list of source files is passed explicitly (via OPENJ9_ONLY_JAVADOC_FILES)
+# to the javadoc tool on the command line and used as dependencies
+# (via OPENJ9_ONLY_JAVADOC_FILELIST) to identify when the javadoc needs
+# regenerating.
+
+OPENJ9_ONLY_FILELIST := $(shell $(FIND) $(SUPPORT_OUTPUTDIR)/j9jcl_sources -type f)
+OPENJ9_ONLY_JAVADOC_FILELIST := $(shell $(FIND) $(SUPPORT_OUTPUTDIR)/j9jcl_sources -type f  '(' -path '*/openj9.*/*.java' -o -path '*/jdk.management/*.java' ')')
+OPENJ9_ONLY_JAVADOC_FILES := $(SUPPORT_OUTPUTDIR)/openj9-docs/_javadoc_openj9_files.txt
+
+# Set up the javadoc command line options.
+# Use  -linkoffline to redirect the links to the documentation for the classes
+# from the openjdk project source files to the# online Oracle documentation.
+# See the  javadoc documentation for details of the linkoffline option - e.g.
+# https://docs.oracle.com/en/java/javase/14/javadoc/javadoc.html
+# Deferred expansion is required for variables because this file is included
+# by make/Docs.gmk before variables such as JAVADOC_OPTIONS and JAVADOC_TAGS
+# have been defined.
+OPENJ9_ONLY_JAVADOC_OPTIONS = $(filter-out --expand-requires transitive,$(JAVADOC_OPTIONS))
+OPENJ9_ONLY_JAVADOC_OPTIONS += -linkoffline $(JAVASE_BASE_URL)/ $(DOCS_OUTPUTDIR)/api/
+OPENJ9_ONLY_JAVADOC_OPTIONS += $(JAVADOC_TAGS)
+OPENJ9_ONLY_JAVADOC_OPTIONS += --module-source-path $(OPENJ9_STUBS_DIR)$(PATH_SEP)$(MODULES_SOURCE_PATH)
+OPENJ9_ONLY_JAVADOC_OPTIONS += -Xdoclint:all,$(call CommaList, $(addprefix -, $(JAVADOC_DISABLED_DOCLINT)))
+
+OPENJ9_DOC_TITLE = $(JDK_LONG_NAME)<br>Version $(VERSION_SPECIFICATION)
+OPENJ9_WINDOW_TITLE = $(subst &amp;,&,$(JDK_SHORT_NAME))
+OPENJ9_HEADER_TITLE = <div $(HEADER_STYLE)><strong>$(JDK_SHORT_NAME)</strong></div>
+
+OPENJ9_ONLY_JAVADOC_OPTIONS += -doctitle '$(OPENJ9_ONLY_JDK_NAME)'
+OPENJ9_ONLY_JAVADOC_OPTIONS += -windowtitle '$(OPENJ9_ONLY_JDK_NAME)'
+OPENJ9_ONLY_JAVADOC_OPTIONS += -header '<div $(HEADER_STYLE)><strong>$(OPENJ9_ONLY_JDK_NAME)</strong></div>'
+OPENJ9_ONLY_JAVADOC_OPTIONS += -bottom '$(OPENJ9_JAVADOC_BOTTOM)'
+
+ifneq ($(LOG_LEVEL), trace)
+  OPENJ9_ONLY_JAVADOC_OPTIONS += -quiet
+else
+  OPENJ9_ONLY_JAVADOC_OPTIONS += -verbose
+endif
+
+OPENJ9_ONLY_JAVADOC_JAVA_ARGS = -Djava.awt.headless=true -Dextlink.spec.version=$(VERSION_SPECIFICATION) -Djspec.version=$(VERSION_SPECIFICATION)
+OPENJ9_ONLY_JAVADOC_CMD = $(JAVA) $(OPENJ9_ONLY_JAVADOC_JAVA_ARGS) $(NEW_JAVADOC)
+
+OPENJDK_MODULE_REFERENCE_TEXT := \
+  These are the OpenJ9 extensions to this module. Refer to <a href="$(JAVASE_BASE_URL)">$(JAVASE_BASE_URL)</a> for the reference implementation documentation.
+
+MODULE_DESCRIPTION_FIRST_LINE := <section class="module-description" id="module\.description">
```

Comment:
This pattern doesn't match anything; instead, I see:

```
<section class="moduleDescription">
<!-- ============ MODULE DESCRIPTION =========== -->
<a id="module.description">
<!--   -->
</a>
```
The replacement should probably use the same class.

Reply:
Thank you for spotting this.  The code was copied from the jdk (head) repo which generates slightly different html.
I've updated the PR (and checked the jdk.management module description which was incorrect).